### PR TITLE
Fixed vanilla not accounting for spritebatch modification in held projectile drawing

### PR DESCRIPTION
**Pull request by ghost**
This solves things such as the auric trail glow bug with certain weapons. Done here instead of in tModLoader due to agreement with Ozzatron that it would be unlikely to be resolved with any speed/at all, and is a quick and easy fix.

### CODE REVIEW

**Review by Ozzatron — DISMISSED** [review]
Simple and effective. Looks good.

File: ILEditing/VanillaBugfixingILChanges.cs [hunk 1]
```diff
@@ -41,5 +44,20 @@ private static void FixSplittingWormBannerDrops(ILContext il)
...
+        private void On_PlayerDrawLayers_DrawHeldProj(On_PlayerDrawLayers.orig_DrawHeldProj orig, PlayerDrawSet drawinfo, Projectile proj)
+        {
+            orig(drawinfo, proj);
+
+            // Vanilla uses a worse quality sampler state for mounts when moving for some reason. Really couldn't say why.
+            var sampler = (drawinfo.drawPlayer.mount.Active && drawinfo.drawPlayer.fullRotation != 0f) ? LegacyPlayerRenderer.MountedSamplerState : Main.DefaultSamplerState;
+
+            // Restart the spritebatch, to ensure that modifications made to it are properly restored.
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, sampler, DepthStencilState.None, Main.Rasterizer, null, Main.GameViewMatrix.TransformationMatrix);
```
Comment (steviegt6) [inline ILEditing/VanillaBugfixingILChanges.cs]: Wrap this in a try-catch; other mods may implement a similar fix.

**Review by steviegt6 — CHANGES_REQUESTED** [review]

**Comment by Ozzatron** [comment]
This bugfix is still waiting for its try catch. I'd like to get it in for Calamity 2.0.3.008 and am willing to delay the release a bit to do so. @toasty599 can you add that error handling?

**Comment by steviegt6** [comment]
Ideally either don't throw anything once caught or log it once so modders know an exception was thrown.

**Review by Ozzatron — DISMISSED** [review]
Looks like a proper try-catch, logged once.

**Review by steviegt6 — DISMISSED** [review]

**Pull request merged**